src/functor.jl

```diff
@@ -253,11 +300,10 @@ Foo(Bar([1, 2, 3]), (40, 50, Bar(Foo(6, 7))))
 ```
 """
 function fmap(f, x; exclude = isleaf, walk = _default_walk, cache = IdDict())
-  haskey(cache, x) && return cache[x]
-  y = exclude(x) ? f(x) : walk(x -> fmap(f, x, exclude = exclude, walk = walk, cache = cache), x)
-  cache[x] = y
-
-  return y
+  return fold(x; cache, walk, isleaf = exclude) do node
+    !exclude(node) && return node
```

Comment:
Shouldn't this be
```suggestion
    exclude(node) && return node
```

Reply:
Can I suggest just this?
```
exclude(node) ? f(node) : node
```
The `&& return` pattern makes more sense when there is real work done below.

Reply:
Also, I retract my suggestion which was wrong, fine with Michael's above.

Reply:
Good catch, I think this was a copy-paste from fcollect.